src/audio/smart_amp/smart_amp.c

```diff
@@ -650,8 +650,17 @@ static int smart_amp_copy(struct comp_dev *dev)
 
 static int smart_amp_reset(struct comp_dev *dev)
 {
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
 	comp_dbg(dev, "smart_amp_reset()");
 
+	sad->process = NULL;
+	sad->in_channels = 0;
+	sad->out_channels = 0;
+	memset(sad->mod_handle, 0, sizeof(sad->mod_handle));
+	memset(&sad->config, 0, sizeof(sad->config));
```

Comment:
we don't want to overwrite params from userspace, just variables from params/prepare/copy steps

Reply:
Fixed. Thanks!